osf/models/metaschema.py

```diff
@@ -148,6 +149,43 @@ def validate_metadata(self, metadata, reviewer=False, required_fields=False):
             raise ValidationValueError(e.message)
         return
 
+    def validate_registration_responses(self, registration_responses, required_fields=False):
+        """
+        Validates registration_responses against the cached jsonschema on the RegistrationSchema.
+        The `title` of the question is stashed under the description for the particular question property
+        for forumulating a more clear error response.
+        """
+        validation_schema = self.registration_responses_jsonschema
+        if not required_fields and validation_schema.get('required', None):
+            del validation_schema['required']
```

Comment:
We don't want to enforce the presence of required fields when the DraftRegistration is being edited, we just need all the required fields present by the time we create the registration itself.  

Here, I just delete the required bit from the jsonschema before validating.